### PR TITLE
Pipeline cleanups miscs

### PR DESCRIPTION
**Pull request by mbouron**

### CODE REVIEW

File: libnopegl/src/backends/gl/pipeline_gl.c [hunk 1]
```diff
@@ -471,8 +427,11 @@ static int pipeline_graphics_init(struct pipeline *s)
     if (gl->features & NGLI_FEATURE_GL_VERTEX_ARRAY_OBJECT) {
         ngli_glGenVertexArrays(gl, 1, &s_priv->vao_id);
         ngli_glBindVertexArray(gl, s_priv->vao_id);
-        init_vertex_attribs(s, gl);
+    } else if (gl->features & NGLI_FEATURE_GL_OES_VERTEX_ARRAY_OBJECT) {
+        ngli_glGenVertexArraysOES(gl, 1, &s_priv->vao_id);
+        ngli_glBindVertexArrayOES(gl, s_priv->vao_id);
     }
```
Comment (ubitux) [inline libnopegl/src/backends/gl/pipeline_gl.c]: can you add the else assert in this `pipeline_graphics_init` function (and maybe in `bind_vertex_attribs`), like you did in the `ngli_pipeline_gl_update_attribute` function?

File: libnopegl/src/backends/gl/pipeline_gl.c [hunk 2]
```diff
@@ -604,15 +563,19 @@ int ngli_pipeline_gl_update_attribute(struct pipeline *s, int32_t index, const s
...
+    const struct buffer_gl *buffer_gl = (const struct buffer_gl *)buffer;
+    if (gl->features & NGLI_FEATURE_GL_VERTEX_ARRAY_OBJECT)
         ngli_glBindVertexArray(gl, s_priv->vao_id);
-        ngli_glBindBuffer(gl, GL_ARRAY_BUFFER, buffer_gl->id);
-        ngli_glVertexAttribPointer(gl, location, size, GL_FLOAT, GL_FALSE, stride, (void*)(uintptr_t)(attribute_binding->desc.offset));
-    }
+    else if (gl->features & NGLI_FEATURE_GL_OES_VERTEX_ARRAY_OBJECT)
+        ngli_glBindVertexArrayOES(gl, s_priv->vao_id);
+    else
+        ngli_assert(0);
```
Comment (ubitux) [inline libnopegl/src/backends/gl/pipeline_gl.c:575]: Why not call just `bind_vertex_attribs` instead here? (or remove it)

File: libnopegl/src/backends/gl/pipeline_gl.c [hunk 3]
```diff
@@ -713,8 +676,6 @@ void ngli_pipeline_gl_draw(struct pipeline *s, int nb_vertices, int nb_instances
     else
         ngli_glDrawArrays(gl, gl_topology, 0, nb_vertices);
 
-    unbind_vertex_attribs(s, gl);
```
Comment (ubitux) [inline libnopegl/src/backends/gl/pipeline_gl.c:716]: How is the unbind related to this commit?

File: libnopegl/src/backends/vk/pipeline_vk.c [hunk 4]
```diff
@@ -602,6 +609,9 @@ static VkResult create_pipeline(struct pipeline *s)
     } else {
         ngli_assert(0);
     }
+
+    s_priv->pipeline_bind_point = bind_point_map[s->type];
```
Comment (ubitux) [inline libnopegl/src/backends/vk/pipeline_vk.c:613]: do you plan to unify the codepaths calling `vkCmdBindPipeline` or is there another rationale here?

File: libnopegl/src/backends/gl/pipeline_gl.c (hunk 1 above)
Comment (mbouron) [reply libnopegl/src/backends/gl/pipeline_gl.c]: Done in both commits.

File: libnopegl/src/backends/gl/pipeline_gl.c (hunk 3 above)
Comment (mbouron) [reply libnopegl/src/backends/gl/pipeline_gl.c:716]: `The unbind_vertex_attribs()` was only there for the codepath not supporting VAOs.

File: libnopegl/src/backends/vk/pipeline_vk.c (hunk 4 above)
Comment (mbouron) [reply libnopegl/src/backends/vk/pipeline_vk.c:613]: I plan to unify the codepaths later on.

File: libnopegl/src/backends/gl/pipeline_gl.c (hunk 2 above)
Comment (mbouron) [reply libnopegl/src/backends/gl/pipeline_gl.c:577]: The next PR will rework these functions, adding code to `bind_vertex_attributes()`, removing entirely ngli_pipeline_update_attributes().

File: libnopegl/src/backends/gl/pipeline_gl.c (hunk 3 above)
Comment (ubitux) [reply libnopegl/src/backends/gl/pipeline_gl.c:716]: Can you mention that in the commit description?

File: libnopegl/src/backends/vk/pipeline_vk.c (hunk 4 above)
Comment (ubitux) [reply libnopegl/src/backends/vk/pipeline_vk.c:613]: Can you mention that in the commit description?

File: libnopegl/src/backends/gl/pipeline_gl.c (hunk 3 above)
Comment (mbouron) [reply libnopegl/src/backends/gl/pipeline_gl.c:716]: Done.

**Review by ubitux — APPROVED** [review]

**Pull request merged**